File: OWASP_scanning/quick_scan.py

import requests
import socket
import ssl
from urllib.parse import urlparse

TARGET_URL = "https://myfiles1.thepihouse.my.id/"

def check_headers_and_config(url):
    print(f"Scanning {url}...")
    try:
        response = requests.get(url, verify=True, timeout=10)
        headers = response.headers
        
        report = ["# OWASP Quick Scan Report",f"Target: {url}", ""]
        
        # 1. Security Headers
        security_headers = {
            "Strict-Transport-Security": "Missing (HSTS ensures HTTPS usage)",
            "Content-Security-Policy": "Missing (Mitigates XSS)",
            "X-Frame-Options": "Missing (Prevents Clickjacking)",
            "X-Content-Type-Options": "Missing (Prevents MIME sniffing)",
            "Referrer-Policy": "Missing (Controls referrer information)",
            "Permissions-Policy": "Missing (Controls browser features)"
        }
        
        report.append("## Security Headers Analysis")
        for header, advice in security_headers.items():
            if header in headers:
                report.append(f"- [OK] {header}: {headers[header]}")
            else:
                report.append(f"- [FAIL] {header}: {advice}")
        
        # 2. Server Information Disclosure
        report.append("\n## Information Disclosure")
        if "Server" in headers:
            report.append(f"- [WARN] Server Header Present: {headers['Server']} (Reveal potential tech stack)")
        if "X-Powered-By" in headers:
            report.append(f"- [WARN] X-Powered-By Header Present: {headers['X-Powered-By']}")
            
        # 3. Cookie Attributes
        report.append("\n## Cookie Security")
        if response.cookies:
            for cookie in response.cookies:
                secure = "Secure" if cookie.secure else "Not Secure"
                httponly = "HttpOnly" if cookie.has_nonstandard_attr("HttpOnly") or cookie.secure else "HttpOnly status unknown" # requests cookiejar handling varies
                # Checking HttpOnly manually is tricky with requests CookieJar, usually better to inspect specific attributes
                # Simplified check:
                report.append(f"- Cookie: {cookie.name}")
                if not cookie.secure:
                    report.append(f"  - [FAIL] Missing 'Secure' flag")
        else:
             report.append("- No cookies found")

        # 4. HTTPS (Basic Check)
        report.append("\n## Transport Security")
        if url.startswith("https"):
            report.append("- [OK] Using HTTPS")
            # Tries to connect to HTTP port 80 to see if it redirects
            try:
                http_url = url.replace("https://", "http://")
                r_http = requests.get(http_url, allow_redirects=False, timeout=5)
                if r_http.status_code in [301, 302, 307, 308] and r_http.headers.get('Location', '').startswith("https"):
                    report.append("- [OK] HTTP redirects to HTTPS")
                else:
                    report.append("- [FAIL] HTTP does not redirect to HTTPS (Status: {r_http.status_code})")
            except Exception as e:
                report.append(f"- [WARN] HTTP check failed: {e}")
                
        # Write Report
        with open("OWASP_scanning/quick_report.md", "w") as f:
            f.write("\n".join(report))
        print("Quick scan completed. Report saved to OWASP_scanning/quick_report.md")

    except Exception as e:
        print(f"Error during scan: {e}")

if __name__ == "__main__":
    check_headers_and_config(TARGET_URL)

File: OWASP_scanning/quick_scan.js

const fs = require('fs');
const https = require('https');
const http = require('http');

const TARGET_URL = "https://myfiles1.thepihouse.my.id/";
const REPORT_FILE = "OWASP_scanning/quick_report.md";

async function checkHeaders(url) {
    console.log(`Scanning ${url}...`);

    try {
        const response = await fetch(url, { method: 'GET' });
        const headers = response.headers;

        let report = [
            "# OWASP Quick Scan Report",
            `Target: ${url}`,
            "",
            "## Security Headers Analysis"
        ];

        const securityHeaders = {
            "strict-transport-security": "Missing (HSTS ensures HTTPS usage)",
            "content-security-policy": "Missing (Mitigates XSS)",
            "x-frame-options": "Missing (Prevents Clickjacking)",
            "x-content-type-options": "Missing (Prevents MIME sniffing)",
            "referrer-policy": "Missing (Controls referrer information)",
            "permissions-policy": "Missing (Controls browser features)"
        };

        for (const [header, advice] of Object.entries(securityHeaders)) {
            if (headers.has(header)) {
                report.push(`- [OK] ${header}: ${headers.get(header)}`);
            } else {
                report.push(`- [FAIL] ${header}: ${advice}`);
            }
        }

        report.push("\n## Information Disclosure");
        if (headers.has("server")) {
            report.push(`- [WARN] Server Header Present: ${headers.get("server")} (Reveal potential tech stack)`);
        }
        if (headers.has("x-powered-by")) {
            report.push(`- [WARN] X-Powered-By Header Present: ${headers.get("x-powered-by")}`);
        }

        report.push("\n## Transport Security");
        if (url.startsWith("https")) {
            report.push("- [OK] Using HTTPS");
        } else {
            report.push("- [FAIL] Not using HTTPS");
        }

        // Write report
        fs.writeFileSync(REPORT_FILE, report.join("\n"));
        console.log(`Quick scan completed. Report saved to ${REPORT_FILE}`);

    } catch (error) {
        console.error("Error during scan:", error);
    }
}

checkHeaders(TARGET_URL);
